### ISSUE TITLE
Hey [couple suggestions + an issue]

### ISSUE THREAD

**Issue by JourneyOver** [Meidor/keywordblocker#21]
hey @Meidor recently found your extension when on the search for a backup to the  [Video Blocker Extension](https://chrome.google.com/webstore/detail/video-blocker/jknkjnpcbbgcbdbaampbjlhkcghmgfhk) it looks pretty great so far though I do have a few suggestions and an issue.

First things first the issue.

No matter what I add in to the blocked keyword list it still shows results with said keywords on the front page. I still have the old youtube layout (no clue on when I'll be getting the new youtube layout).

This is the error in the console -- `
Uncaught ReferenceError: videolink is not defined
    at chrome-extension://pbgacppomjfpheddhifkdkklddnolnpg/scripts/contentscript.js:62:40
    at Array.forEach (native)
    at examineFrontpage (chrome-extension://pbgacppomjfpheddhifkdkklddnolnpg/scripts/contentscript.js:56:16)
    at MutationObserver.<anonymous> (chrome-extension://pbgacppomjfpheddhifkdkklddnolnpg/scripts/contentscript.js:145:13)
(anonymous) @ contentscript.js:62
examineFrontpage @ contentscript.js:56
(anonymous) @ contentscript.js:145
VM887:1 GET https://googleads.g.doubleclick.net/pagead/id net::ERR_BLOCKED_BY_CLIENT
(anonymous) @ VM887:1
Xf @ base.js:366
g.I @ base.js:354
(anonymous) @ /yts/jsbin/www-en_US-vflfKKoYP/common.js:482
g.Oc @ base.js:174
ks @ /yts/jsbin/www-en_US-vflfKKoYP/common.js:479
(anonymous) @ /yts/jsbin/www-en_US-vflfKKoYP/common.js:491
(anonymous) @ base.js:293`

Things seem to work just fine doing a search though as it blocks all videos with keywords in the block list.

As for suggestions.

as noted by [Issue 19](https://github.com/Meidor/keywordblocker/issues/19) gotta get this in asap (as well as [Issue 20](https://github.com/Meidor/keywordblocker/issues/20) it would really up this extension to being a whole lot better vs the video blocker extension (especially since the dev who made that extension disappeared off the face of the planet) four things for this though

1. Have normal channel blocking that'll block a channel from appearing outright.

2. Have a wildcard channel blocking method as well. (ie - "difference is that a wildcard item will hide videos by channels that contain the wildcard in their display name. Whereas the normal channel item needs to 100% match the uploaders channel name." )

3. When you implement channel blocking into your extension, have the extension block playlists from said channel from showing up in the search as well, currently video blocker is kinda iffy on this and sometimes works and sometimes doesn't, so if you could get this implemented in your extension it would be great

4. Have the ability to block a video by keyword for only certain channels while allowing videos/channel with said keyword to still show up (ie - block the word "BLOCKTHIS" on the "YOUTUBECHANNELTOBLOCK" channel, but not block any other users using the keyword "BLOCKTHIS".), this was a suggestion on the video blocker extension not one of my own but may be a useful feature.

as noted in [Issue 18](https://github.com/Meidor/keywordblocker/issues/18) once you get channel blocking implemented as well, I'd love to see the ability to right click on a video and adding the channel/video to the block list. something like so -- ![Image](https://github.com/JourneyOver/Youtube-Video-Blocker/blob/master/images/screenshot3.png?raw=true) which when clicked would add the channel name to the blocked list.

for now these are the few suggestions that I have.

**Comment by Meidor** [comment]
Thank you for the excellent feedback, I found the cause of the issue you reported and will fix it in the new update. I think all your suggestions are excellent and will see what I can implement today. I'll keep you posted!

**Comment by Meidor** [comment]
I've implemented most of your suggestions.
I will add the block keyword on specified channel only feature as a seperate to-do issue.
I've yet to implement the right click to add channel to blocklist feature but I did add a feature to add selected text as a blocked (wildcard) keyword or channel.

If you have anymore suggestions I would love te hear!  If you would be so kind to make seperate issues per suggestion, that makes it easy to use as a to-do list.

Kind Regards,

Rik
